crates/ruma-events/src/room/message/audio.rs

```diff
@@ -12,8 +13,18 @@ use crate::room::{EncryptedFile, MediaSource};
 #[serde(tag = "msgtype", rename = "m.audio")]
 pub struct AudioMessageEventContent {
     /// The textual representation of this message.
+    /// If the `filename` field is set and differs from `body`, `body` should be interpreted as a
+    /// user-written media caption.
```

Comment:
It is not clear that it is the filename if it is not a caption.

Maybe something like this (the empty line at the start is on purpose):

```suggestion
    ///
    /// If the `filename` field is not set or has the same value, this is the filename of the
    /// uploaded file. Otherwise, this should be interpreted as a user-written media caption.
```

And the same docs for all other types.

Reply:
Pushed.